src/sage/combinat/lp_algebra_seed.py

```diff
@@ -0,0 +1,1219 @@
+            (x1, x2, x3)
+            sage: S = LpaSeed({x1: 1 + x2, x2: 1 + x1})
+            sage: print(S.variable_class())
+            [(x1 + x2 + 1)/(x1*x2), (x2 + 1)/x1, (x1 + 1)/x2, x2, x1]
+            sage: S = LpaSeed({x1: 1 + x2 + x3, x2: 1 + x1 + x3, x3: 1 + x1 + x2})
+            sage: print(S.variable_class())
+            [(x1 + x2 + x3 + 1)/(x1*x2*x3), (x2 + x3 + 1)/x1, (x1 + x3 + 1)/x2, (x1 + x2 + 1)/x3, x3, x2, x1]
+        """
+
+        var_iter = self.variable_class_iter(depth=depth)
+        return sorted(var_iter)
+
+    def is_equivalent(self, other):
+        r"""
+        Return whether ``self`` and ``other`` are equivalent as LP seeds.
+        Two seeds are equivalent if and only if there is a permutation of the
+        cluster variables of one seed to get the cluster variables of the other
+        seed, up to unit multipliers. Note we also overload equality to
+        equivalence.
+
+        INPUT:
+
+        - ``other`` -- ``LpaSeed``; the seed which we are comparing ``self`` to.
+
+        EXAMPLES:
+
+        Mutating this rank two example five times yields an equivalent seed::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LpaSeed({x1: 1 + x2, x2: 1 + x1})
+            sage: print(S.is_equivalent(S.mutate([0,1,0,1,0], inplace=False)))
+            True
+        """
+        if not isinstance(other, LpaSeed):
+            raise ValueError('%s is not a seed!' % (other))
+
+        if self._rank != other._rank:
+            return False
+
+        n = self._rank
+        L = []
+        for i in range(n):
+            for j in range(n):
+                x = other._cluster_vars[j]
+                y = self._cluster_vars[i]
+                t = x / y
+                if t.numerator().is_unit() and t.denominator().is_unit():
+                    L.append(j)
+        return len(L) == n
+
+    def exchange_graph(self):
+        r"""
+        Return the exchange graph of ``self``.
+
+        EXAMPLES:
+
+        We work out the exchange graph for a rank-two example::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: LpaSeed({x1: 1 + x2, x2: 1 + x1}).exchange_graph()
+            Graph on 5 vertices
+
+        .. PLOT::
+
+            var('x1, x2')
+            G = LpaSeed({x1: 1 + x2, x2: 1 + x1}).exchange_graph()
+            sphinx_plot(G)
+
+        A rank three example::
+
+            sage: var('x1, x2, x3')
+            (x1, x2, x3)
+            sage: LpaSeed({x1: 1 + x2 + x3, x2: 1 + x1 + x3, x3: 1 + x1 + x2}).exchange_graph()
+            Graph on 10 vertices
+
+        .. PLOT::
+
+            var('x1, x2, x3')
+            G = LpaSeed({x1: 1 + x2 + x3, x2: 1 + x1 + x3, x3: 1 + x1 + x2}).exchange_graph()
+            sphinx_plot(G)
+        """
+
+        # initialise
+
+        covers = []
+        n = self.rank()
+        stack = [self]
+        known_seeds = []
+
+        # do a depth-first search
+        while stack != []:
```

Comment:
```suggestion
        while stack:
```
This is slightly faster.